lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1637,6 +1637,51 @@ class DecomposeAtenIndexPutOp : public OpRewritePattern<AtenIndexPutOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.index_add` op into `valsem.aten.index_put_impl` op. with
+// accumulate set to True
+class DecomposeAtenIndexAddOp : public OpRewritePattern<AtenIndexAddOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenIndexAdd op,
+                                PatternRewriter &rewriter) const override {
+
+    Value input = op.self();
+    Value updates = op.values();
```

Comment:
There is no `values` in `AtenIndexAddOp`. You can see the available fields by looking at the ODS generated:

```
let arguments = (ins
    AnyTorchTensorType:$self,
    Torch_IntType:$dim,
    AnyTorchTensorType:$index,
    AnyTorchTensorType:$source,
    AnyTorchScalarType:$alpha
  );
```

Reply:
Yeah, although the convention in torch-mlir is to call the variable after the op operand name, so I would call it `Value source = op.source();`.